src/VisualStudio/Core/Def/Packaging/PackageInstallerServiceFactory.cs

```diff
@@ -45,36 +52,50 @@ namespace Microsoft.VisualStudio.LanguageServices.Packaging
     [ExportWorkspaceService(typeof(IPackageInstallerService)), Shared]
     internal partial class PackageInstallerService : AbstractDelayStartedService, IPackageInstallerService, IVsSearchProviderCallback
     {
-        private readonly object _gate = new object();
         private readonly VisualStudioWorkspaceImpl _workspace;
         private readonly SVsServiceProvider _serviceProvider;
+        private readonly Shell.IAsyncServiceProvider _asyncServiceProvider;
         private readonly IVsEditorAdaptersFactoryService _editorAdaptersFactoryService;
 
         private readonly Lazy<IVsPackageInstallerServices> _packageInstallerServices;
         private readonly Lazy<IVsPackageInstaller2> _packageInstaller;
         private readonly Lazy<IVsPackageUninstaller> _packageUninstaller;
         private readonly Lazy<IVsPackageSourceProvider> _packageSourceProvider;
 
-        private JoinableTask<ImmutableArray<PackageSource>?> _packageSourcesAsync;
-        private IVsPackage _nugetPackageManager;
-
-        private CancellationTokenSource _tokenSource = new CancellationTokenSource();
+        private IVsPackage? _nugetPackageManager;
 
-        // We keep track of what types of changes we've seen so we can then determine what to
-        // refresh on the UI thread.  If we hear about project changes, we only refresh that
-        // project.  If we hear about a solution level change, we'll refresh all projects.
-        private bool _solutionChanged;
-        private readonly HashSet<ProjectId> _changedProjects = new HashSet<ProjectId>();
+        /// <summary>
+        /// Used to keep track of what types of changes we've seen so we can then determine what to refresh on the UI
+        /// thread.  If we hear about project changes, we only refresh that project.  If we hear about a solution level
+        /// change, we'll refresh all projects.
+        /// </summary>
+        /// <remarks>
+        /// <c>solutionChanged == true iff changedProject == null</c> and <c>solutionChanged == false iff changedProject
+        /// != null</c>. So technically having both values is redundant.  However, i like the clarity of having both.
+        /// </remarks>
```

Comment:
Stale comment?

Reply:
➡️ This is referring to the members of the tuple type, not the `_solutionChanged` field from the initial code.

Reply:
yup!